packages/itwinui-react/src/core/ThemeProvider/ThemeProvider.tsx

```diff
@@ -261,22 +242,26 @@ const useParentThemeAndContext = (rootElement: HTMLElement | null) => {
       return;
     }
 
+    // helper function that updates state to match data attributes from closest root
+    const synchronizeTheme = () => {
+      setParentTheme(closestRoot?.getAttribute('data-iui-theme') as ThemeType);
+      setParentHighContrastState(
+        closestRoot?.getAttribute('data-iui-contrast') === 'high',
+      );
+    };
+
     // set theme for initial mount
-    setParentTheme(closestRoot?.getAttribute('data-iui-theme') as ThemeType);
+    synchronizeTheme();
```

Comment:
Is this needed? Because aren't we initializing `parentThemeState` and `parentHighContrastState` as a param in the `React.useState(initialValue)`?

Reply:
no, the initial value of state uses context.